autoinit/estimators/tf_op_lambda.py

```diff
@@ -0,0 +1,165 @@
+
+# Copyright (C) 2021-2022 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# AutoInit Software in commercial settings.
+#
+# END COPYRIGHT
+
+import logging
+import math
+
+from typing import List
+
+from numpy import average, prod
+
+# TF uses a complicated LazyLoader that pylint cannot properly comprehend.
+# See https://stackoverflow.com/questions/65271399/vs-code-pylance-pylint-cannot-resolve-import
+import tensorflow.keras as tfkeras # pylint: disable=import-error
+
+from autoinit.estimators.estimate_layer_output_distribution \
+    import LayerOutputDistributionEstimator
+
+
+class TFOpLambdaOutputDistributionEstimator(LayerOutputDistributionEstimator):
+    """
+    TensorFlow wraps API calls (like tf.nn.relu) in a TFOpLambda Layer object.
+    Similarly, sometimes users wrap arbitrary functions in Lambda layers.
+    In these cases, the best we can do is to try to infer the layer operation
+    from the layer name.  Often the layer names will have suffixes, like tf.reshape_3,
+    so we just check whether these API calls are in the layer name.  This class is a
+    workaround.  If possible, API calls should be replaced with proper Layer objects.
+    """
```

Comment:
This whole implementation is a bit of a hack, but with API calls we don't have access to the usual Layer attributes.  Inferring functionality from the name seems like the least bad approach.

Reply:
Why didn't you need this until now?

Reply:
The architectures I was using only used Layers.  When I applied AutoInit to CoAtNet there were a lot of these API calls.

Reply:
makes sense; thanks